Husky integration

treykris opened the pull request:
Description
-----------
- Added Husky for git hooks to run lint before committing and to run our end-to-end tests before pushing to GitHub

Developer self-review checklist
-------------------------------
- [x] Potentially confusing code has been explained with comments
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] All edge cases have been addressed

jkeat commented on run-cypress.sh:
```diff
@@ -0,0 +1,8 @@
+else
+    echo "Please verify that your client and server are running locally before pushing to GitHub."
+    exit 1
+fi
```
Could you add a folder called `bin/` and put this inside of it? Similar to the server repo, where all the bash scripts are in a `bin/` dir

treykris commented:
@jkeat 

Updated shell path and placed it inside the bin folder

treykris commented:
@jkeat,

> When I git pushed, it ran all the tests successfully but didn't end up doing the actual git push.. Any ideas?

Did all of the tests pass? If a test fails I believe it throws an exit code 1 and then refuses to push. Have to add the `--no-verify` to the git command in order to bypass husky - `git push --no-verify origin husky-integration`

> I wonder if the test suite could run a bit faster if we disabled the video recording and/or screenshots? (could also play with the videoCompression value) https://docs.cypress.io/guides/references/configuration.html#Videos

Good idea, I just updated it! The tests still run ridiculously slow, especially the session tests that I added here https://github.com/UPchieve/web/pull/339
